src/context/AppProvider/AppContext.tsx

```diff
@@ -137,169 +137,166 @@ export const AppProvider = ({ children }: { children: React.ReactNode }) => {
       return requestedAccountIds.length > 0
     })()
 
+    let chainIds = new Set(
+      supportedChains.filter(chainId => {
+        return walletSupportsChain({
+          chainId,
+          wallet,
+          isSnapInstalled,
+          checkConnectedAccountIds: false, // don't check connected account ids, we're detecting runtime support for chains
+        })
+      }),
+    )
+    if (!chainIds.size) return
     ;(async () => {
-      try {
```

Comment:
async flow was absolutely off: try/finally doesn't work here and the finally block would kick in before promise settles. Chained a `.then()` instead since we use `Promise.allSettled()` everywhere, although a `.finally()` could also work if we want some paranoia.